node/node.go

```diff
@@ -426,18 +432,29 @@ func (node *Node) validateShardBoundMessage(
 		return nil, nil, errors.WithStack(errNotRightKeySize)
 	}
 
-	key, err := bls_cosi.BytesToBLSPublicKey(
-		senderPubKeyViaWire,
-	)
+	kkey := string(senderPubKeyViaWire)
+	key, err, _ := node.blsPubKeyCache.Do(kkey, func() (interface{}, error) {
+		k, e := bls_cosi.BytesToBLSPublicKey(senderPubKeyViaWire)
```

Comment:
how about moving the caching inside `BytesToBLSPublicKey`? then all the fbft message parsing can also take advantage. e.g., `ParseFBFTMessage` may speedup too.

Reply:
good suggestion, let me try it.